aws/resource_aws_appconfig_deployment.go

```diff
@@ -0,0 +1,216 @@
+	deploymentStrategyID := aws.String(d.Get("deployment_strategy_id").(string))
+	configProfileID := aws.String(d.Get("configuration_profile_id").(string))
+	deploymentDescription := aws.String(d.Get("description").(string))
+	configVersion := aws.String(d.Get("configuration_version").(string))
+
+	input := &appconfig.StartDeploymentInput{
+		ApplicationId:          applicationID,
+		EnvironmentId:          environmentID,
+		DeploymentStrategyId:   deploymentStrategyID,
+		ConfigurationProfileId: configProfileID,
+		ConfigurationVersion:   configVersion,
+		Description:            deploymentDescription,
+		Tags:                   keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().AppconfigTags(),
+	}
+
+	var deploy *appconfig.StartDeploymentOutput
+	var err error
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
+		deploy, err = conn.StartDeployment(input)
+		if err != nil {
+			if isAWSErr(err, appconfig.ErrCodeConflictException, "") {
+				log.Printf("[DEBUG] Retrying AppConfig Deployment creation: %s", err)
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		deploy, err = conn.StartDeployment(input)
+	}
+
+	if err != nil {
+		return fmt.Errorf("Creating AppConfig Deployment failed: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s-%s-%d", aws.StringValue(applicationID), aws.StringValue(environmentID), aws.Int64Value(deploy.DeploymentNumber)))
+	d.Set("deployment_number", aws.Int64Value(deploy.DeploymentNumber))
+
+	return resourceAwsAppconfigDeploymentRead(d, meta)
+}
+
+func resourceAwsAppconfigDeploymentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	applicationID := aws.String(d.Get("application_id").(string))
+	environmentID := aws.String(d.Get("environment_id").(string))
+
+	input := &appconfig.GetDeploymentInput{
+		ApplicationId:    applicationID,
+		EnvironmentId:    environmentID,
+		DeploymentNumber: aws.Int64(int64(d.Get("deployment_number").(int))),
+	}
+
+	output, err := conn.GetDeployment(input)
+
+	if isAWSErr(err, appconfig.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] Appconfig Deployment (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error getting AppConfig Deployment (%s): %s", d.Id(), err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("error getting AppConfig Deployment (%s): empty response", d.Id())
+	}
+
+	currentState := aws.StringValue(output.State)
+	if currentState == appconfig.DeploymentStateRolledBack || currentState == appconfig.DeploymentStateRollingBack {
+		log.Printf("[WARN] Appconfig Deployment (%s) rolled back, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	appID := aws.StringValue(output.ApplicationId)
```

Comment:
can you use the original ones that you passed in? do you expect these values to be changed?

Reply:
AFAIK the IDs should be the same. Deploy number is different since the user doesn't pass it in. It's given to us by AWS when we start the deployment. I figured we might as well take everything from the output instead of taking some stuff from output and some from what the user passed in.